numalogic/udfs/trainer/_rds.py

```diff
@@ -0,0 +1,272 @@
+import logging
+import time
+from typing import Optional
+import pandas as pd
+import hashlib
+from numalogic.config.factory import ConnectorFactory
+from numalogic.connectors import RDSFetcherConf
+from numalogic.tools.exceptions import ConfigNotFoundError, RDSFetcherError
+from numalogic.tools.types import redis_client_t
+from numalogic.udfs._config import PipelineConf
+from numalogic.udfs.entities import TrainerPayload
+from numalogic.udfs._metrics import (
+    FETCH_EXCEPTION_COUNTER,
+    DATAFRAME_SHAPE_SUMMARY,
+    FETCH_TIME_SUMMARY,
+    _increment_counter,
+    _add_summary,
+)
+from numalogic.udfs.trainer._base import TrainerUDF
+from datetime import datetime, timedelta
+import pytz
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_hash_based_query(config_id: str, filter_keys=list[str], filter_values=list[str]):
+    """
+    Calculate the hash value based on the given configuration ID and filter keys/values.
+
+    Args:
+        - config_id (str): The configuration ID.
+        - filter_keys (list[str]): The list of filter keys.
+        - filter_values (list[str]): The list of filter values.
+
+    Returns
+    -------
+    - hash (str): The calculated hash value.
+
+    Raises
+    ------
+    - RDSFetcherError: If the length of filter_keys and filter_values is not equal.
+
+    """
+    if len(filter_keys) != len(filter_values):
+        raise RDSFetcherError("filter_keys and filter_values length are not equal")
+
+    filter_pairs = dict(zip(filter_keys, filter_values))
+    filter_pairs["config_id"] = config_id
+    hash_keys_sorted = sorted(filter_pairs.keys())
+
+    to_be_hashed_list = []
+    for key in hash_keys_sorted:
+        to_be_hashed_list.append(filter_pairs[key].strip())
+    str_to_be_hashed = "".join(to_be_hashed_list)
+    result = hashlib.md5(("".join(str_to_be_hashed)).encode(), usedforsecurity=False)
+    hash = result.hexdigest()
+    _LOGGER.info(
```

Comment:
@gulshan02 these logs are coming your ways :)

Reply:
yes, this would need to be changed to be compliance with struct log that we have. let me check it. Thanks for pointing out